runtime/sample/src/main.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2015-2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
+
+// Parity Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Simple example, illustating usage of runtime wrapper.
+
+use futures::{Future, Stream};
+use parity_runtime::Runtime;
+use tokio::fs::read_dir;
+
+/// Read current directory in future, which is executed in the created runtime
+fn main() {
+	let fut = read_dir(".")
```

Comment:
why not include this in `doctest` instead in the crate root to be part of the documentation?

Reply:
AFAIU the idea was to run this sample as standard cargo example in order to make sure, that public API is declared properly